05/app.js

```diff
@@ -9,20 +9,42 @@ const stats = {
 
 /* tutaj umieść swój kod */
 
+const Listp = document.querySelectorAll("p");
+const Listlink = document.querySelectorAll("[href = '/dolor.html']");
+
+const countLinks = function(e) {
+    e.preventDefault();
+    stats.links["/dolor.html"]+=1;
+}
+    
+const countClicks = function(e) {
+    e.preventDefault();
+    if(e.target === e.currentTarget) {
+        stats.paragraphs.p1+=1;
+        }
+}
+
+Listp.forEach(function(p) {
+    p.addEventListener('click', countClicks)
+});
+
+Listlink.forEach(function(link) {
+    link.addEventListener('click', countLinks);
+});
```

Comment:
Mieliśmy operować tylko na kliku dla `p` - czy kliknięto na `a` można sprawdzić przez propagację i `e.target.getName === 'a'`